test/e2e/autoscale_test.go

```diff
@@ -357,12 +383,89 @@ func assertAutoscaleUpToNumPods(ctx *testContext, curPods, targetPods float64, d
 	}
 }
 
+func assertGracefulScaledown(t *testing.T, ctx *testContext, size int) error {
+	// start x running pods; x == size
+	hostConnMap, err := uniqueHostConnections(t, ctx.names, size)
+	if err != nil {
+		return err
+	}
+
+	// only keep openConnCount connections open for the test
+	openConnCount := size/2
+	deleteHostConnections(hostConnMap, size-openConnCount)
+
+	defer deleteHostConnections(hostConnMap, openConnCount)
+	done := make(chan struct{})
+	go func() error {
+		defer close(done)
+
+		timer := time.NewTicker(2 * time.Second)
+		for _ = range timer.C {
+				readyCount, err := numberOfReadyPods(ctx)
+				if err != nil {
+					return err
+				}
+
+				if int(readyCount) < openConnCount {
+					t.Fatalf("Failed keeping the right number of pods. Ready(%d) != Expected(%d)", int(readyCount), openConnCount)
+				}
+
+				if int(readyCount) == openConnCount {
+					pods, err := allPods(ctx)
+					if err != nil {
+						return err
+					}
+
+					for _, p := range pods {
+						if p.Status.Phase != corev1.PodRunning || p.DeletionTimestamp != nil {
+							continue
+						}
+
+						if _, ok := hostConnMap.Load(p.Name); !ok {
+							t.Fatalf("Failed by keeping the wrong pod %s", p.Name)
+						}
+					}
+					break
+				}
+			}
+			return nil
+	}()
+
+	<-done
+	return nil
+}
+
+func TestGracefulScaledown(t *testing.T) {
+	t.Skip()
```

Comment:
it almost constantly fails without the skip. Hence the `skip`

Reply:
Please add a description to the Skip then: `t.Skip("Skipping because we're in the process of implementing the mechanics to make this test reliably pass`